src/common/media.h

```diff
@@ -0,0 +1,52 @@
+// media.h
+#ifndef _MEDIA_H
+#define _MEDIA_H
+
+#include <inttypes.h>
+
+typedef enum {
+    media_error_OK = 0,    // no error
+    media_error_MOUNT = 1, // error - mounting media - f_mount failed
+} media_error_t;
+
+typedef enum {
+    media_print_state_NONE = 0,
+    media_print_state_PRINTING = 1,
+    media_print_state_PAUSED = 2,
+} media_print_state_t;
+
+#ifdef __cplusplus
+extern "C" {
+#endif //__cplusplus
+
+extern char media_print_filename[128];
+
+extern char media_print_filepath[128];
+
+extern uint8_t media_is_inserted(void);
+
+extern void media_print_start(const char *filepath);
+
+extern void media_print_stop(void);
+
+extern void media_print_pause(void);
+
+extern void media_print_resume(void);
+
+extern media_print_state_t media_print_get_state(void);
+
+extern uint32_t media_print_get_size(void);
+
+extern uint32_t media_print_get_position(void);
+
+extern void media_loop(void);
+
+extern void media_set_inserted(uint8_t inserted);
```

Comment:
Do you disagree with me, that `bool` should be used for boolean values?

Or do you just think, that this can be refactored even further – into two separate functions?

Reply:
I try to avoid `bool` parameters if possible.
I especially do not like `void set_something(bool set);`
I have nothing against `bool `return value or variables.
In this case I think it should be refactored even further.

I would have nothing against
`void media_set_state(some_enum state);`
since it can be extended with more states like disabled.

Reply:
But 

> void media_set_inserted();
> void media_clear_inserted();

seems most clear to me.

Reply:
I understand your point and I even agree, that splitting it is the nicer alternative 👍

I was just surprised you disagreed with me, while the only thing I said is „Use bool for boolean values.“. Do you not agree on using `bool` to represent boolean values?

My only point was, that I too often see boolean values being stored in `uint8_t` instead of `bool`.

Reply:
I am not happy with bool, finally I have chosen solution with enum.